proposals/infra/INF-0002-root-signature-in-clang.md

```diff
@@ -0,0 +1,717 @@
+| - | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
+| | i32 | i32 | i32 | i32 | float | i32 | i32 | i32 | float | float | i32 | i32 | i32 |
+
+
+## Examples
+
+
+* Example root signature string
+
+```
+
+    #define RS "RootFlags( ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT | " \
+              "DENY_VERTEX_SHADER_ROOT_ACCESS), " \
+              "CBV(b0, space = 1, flags = DATA_STATIC), " \
+              "SRV(t0), " \
+              "UAV(u0), " \
+              "DescriptorTable( CBV(b1), " \
+              "                 SRV(t1, numDescriptors = 8, " \
+              "                     flags = DESCRIPTORS_VOLATILE), " \
+              "                 UAV(u1, numDescriptors = unbounded, " \
+              "                     flags = DESCRIPTORS_VOLATILE)), " \
+              "DescriptorTable(Sampler(s0, space=1, numDescriptors = 4)), " \
+              "RootConstants(num32BitConstants=3, b10), " \
+              "StaticSampler(s1)," \
+              "StaticSampler(s2, " \
+              "              addressU = TEXTURE_ADDRESS_CLAMP, " \
+              "              filter = FILTER_MIN_MAG_MIP_LINEAR )"
+
+```
+
+* Define of HLSLEntryRootSignatureAttr
+
+```
+
+    def HLSLEntryRootSignature: InheritableAttr {
+      let Spellings = [GNU<"RootSignature">];
+      let Subjects = Subjects<[HLSLEntry]>;
+      let LangOpts = [HLSL];
+      let Args = [StringArgument<"InputString">, DeclArgument<HLSLRootSignature, "RootSignature", 0, /*fake*/ 1>];
+    }
+
+```
+
+* HLSLEntryRootSignatureAttr in AST.
+
+```
+    HLSLRootSignature 'main.RS'
+      "RootFlags( ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT | DENY_VERTEX_SHADER_ROOT_ACCESS),
+       CBV(b0, space = 1, flags = DATA_STATIC), SRV(t0), UAV(u0),
+       DescriptorTable( CBV(b1), SRV(t1, numDescriptors = 8,flags = DESCRIPTORS_VOLATILE),
+       UAV(u1, numDescriptors = unbounded, flags = DESCRIPTORS_VOLATILE)),
+       DescriptorTable(Sampler(s0, space=1, numDescriptors = 4)),
+       RootConstants(num32BitConstants=3, b10),
+       StaticSampler(s1),
+       StaticSampler(s2, addressU = TEXTURE_ADDRESS_CLAMP, filter = FILTER_MIN_MAG_MIP_LINEAR )"
+
+    HLSLEntryRootSignatureAttr
+      "RootFlags( ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT | DENY_VERTEX_SHADER_ROOT_ACCESS),
+       CBV(b0, space = 1, flags = DATA_STATIC), SRV(t0), UAV(u0),
+       DescriptorTable( CBV(b1), SRV(t1, numDescriptors = 8,flags = DESCRIPTORS_VOLATILE),
+       UAV(u1, numDescriptors = unbounded, flags = DESCRIPTORS_VOLATILE)),
+       DescriptorTable(Sampler(s0, space=1, numDescriptors = 4)),
+       RootConstants(num32BitConstants=3, b10),
+       StaticSampler(s1),
+       StaticSampler(s2, addressU = TEXTURE_ADDRESS_CLAMP, filter = FILTER_MIN_MAG_MIP_LINEAR )"
+       HLSLRootSignature 'main.RS'
+
+```
+It is possible to avoid the root signature string in HLSLEntryRootSignatureAttr
```

Comment:
I'm confused by this statement.  Are you pointing out an optimization here?  Leaving out / adding the root signature string in the printed output?  Is this for diagnostics?

Reply:
I'm pointing out that print the root signature string for HLSLEntryRootSignatureAttr when dump AST is optional for implementation.